text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1607 @@
+}
+
+// A ProfileContainer represents a single profile. It wraps pprof profile with OpenTelemetry specific metadata.
+message ProfileContainer {
+  // A unique identifier for a profile. The ID is a 16-byte array. An ID with
+  // all zeroes is considered invalid.
+  //
+  // This field is required.
+  bytes profile_id = 1;
+
+  // start_time_unix_nano is the start time of the profile.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  //
+  // This field is semantically required and it is expected that end_time >= start_time.
+  fixed64 start_time_unix_nano = 2;
+
+  // end_time_unix_nano is the end time of the profile.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  //
+  // This field is semantically required and it is expected that end_time >= start_time.
+  fixed64 end_time_unix_nano = 3;
+
+  // attributes is a collection of key/value pairs. Note, global attributes
+  // like server name can be set using the resource API. Examples of attributes:
+  //
+  //     "/http/user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36"
+  //     "/http/server_latency": 300
+  //     "abc.com/myattribute": true
+  //     "abc.com/score": 10.239
+  //
+  // The OpenTelemetry API specification further restricts the allowed value types:
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute
+  // Attribute keys MUST be unique (it is not allowed to have more than one
+  // attribute with the same key).
+  repeated opentelemetry.proto.common.v1.KeyValue attributes = 4;
+
+  // dropped_attributes_count is the number of attributes that were discarded. Attributes
+  // can be discarded because their keys are too long or because there are too many
+  // attributes. If this value is 0, then no attributes were dropped.
+  uint32 dropped_attributes_count = 5;
+
+  // Specifies format of the original payload. Common values are defined in semantic conventions. [required if original_payload is present]
+  string original_payload_format = 6;
+
+  // Original payload can be stored in this field. This can be useful for users who want to get the original payload.
+  // Formats such as JFR are highly extensible and can contain more information than what is defined in this spec.
+  // Inclusion of original payload should be configurable by the user. Default behavior should be to not include the original payload.
+  // If the original payload is in pprof format, it SHOULD not be included in this field.
+  // The field is optional, however if it is present `profile` MUST be present and contain the same profiling information.
+  bytes original_payload = 7;
+
+  // This is a reference to a pprof profile. Required, even when original_payload is present.
+  opentelemetry.proto.profiles.v1.alternatives.pprofextended.Profile profile = 8;
+}
+```
+
+<!-- proto1 -->
```

Comment:
What's this comment about?

Reply:
This is for a script that I use to automatically update the protos from [opentelemetry-proto-profile repo](https://github.com/petethepig/opentelemetry-proto/tree/pprof-experiments) in the right place.